library/src/hipsparse_module.f90

```diff
@@ -3153,6 +3170,256 @@ function hipsparseZcsrilu02(handle, m, nnz, descr, csrVal, csrRowPtr, &
+            use iso_c_binding
+            implicit none
+            type(c_ptr), value :: handle
+            integer(c_int), value :: dir
+            integer(c_int), value :: mb
+            integer(c_int), value :: nnzb
+            type(c_ptr), intent(in), value :: descr
+            type(c_ptr), intent(in), value :: bsrVal
+            type(c_ptr), intent(in), value :: bsrRowPtr
+            type(c_ptr), intent(in), value :: bsrColInd
+            integer(c_int), value :: blockDim
+            type(c_ptr), value :: info
+            type(c_ptr), value :: bufferSize
+        end function hipsparseSbsric02_bufferSize
+
+        function hipsparseDbsric02_bufferSize(handle, dir, mb, nnzb, descr, bsrVal, &
+                bsrRowPtr, bsrColInd, blockDim, info, bufferSize) &
+                result(c_int) &
+                bind(c, name = 'hipsparseDbsric02_bufferSize')
+            use iso_c_binding
+            implicit none
+            type(c_ptr), value :: handle
+            integer(c_int), value :: dir
+            integer(c_int), value :: mb
+            integer(c_int), value :: nnzb
+            type(c_ptr), intent(in), value :: descr
+            type(c_ptr), intent(in), value :: bsrVal
+            type(c_ptr), intent(in), value :: bsrRowPtr
+            type(c_ptr), intent(in), value :: bsrColInd
+            integer(c_int), value :: blockDim
+            type(c_ptr), value :: info
+            type(c_ptr), value :: bufferSize
+        end function hipsparseDbsric02_bufferSize
+
+        function hipsparseCbsric02_bufferSize(handle, dir, mb, nnzb, descr, bsrVal, &
+                bsrRowPtr, bsrColInd, blockDim, info, bufferSize) &
+                result(c_int) &
+                bind(c, name = 'hipsparseCbsric02_bufferSize')
+            use iso_c_binding
+            implicit none
+            type(c_ptr), value :: handle
+            integer(c_int), value :: dir
+            integer(c_int), value :: mb
+            integer(c_int), value :: nnzb
+            type(c_ptr), intent(in), value :: descr
+            type(c_ptr), intent(in), value :: bsrVal
+            type(c_ptr), intent(in), value :: bsrRowPtr
+            type(c_ptr), intent(in), value :: bsrColInd
+            integer(c_int), value :: blockDim
+            type(c_ptr), value :: info
+            type(c_ptr), value :: bufferSize
+        end function hipsparseCbsric02_bufferSize
+
+        function hipsparseZbsric02_bufferSize(handle, dir, mb, nnzb, descr, bsrVal, &
+                bsrRowPtr, bsrColInd, blockDim, info, bufferSize) &
+                result(c_int) &
+                bind(c, name = 'hipsparseZbsric02_bufferSize')
+            use iso_c_binding
+            implicit none
+            type(c_ptr), value :: handle
+            integer(c_int), value :: dir
+            integer(c_int), value :: mb
+            integer(c_int), value :: nnzb
+            type(c_ptr), intent(in), value :: descr
+            type(c_ptr), intent(in), value :: bsrVal
```

Comment:
bsrVal is non-const (also in other precisions)

Reply:
Are you referring to the fact that in cusparseSbsric0_bufferSize etc pass bsrVAl by non const? I think this is a bug as bsrVAl is not modified by this function. I made the rocsparse routines pass it const.

Reply:
Looks like we do the same for csric0 even though cusparse uses non-const